draft-ietf-ppm-dap.md

```diff
@@ -1554,14 +1563,26 @@ struct {
 where `prepare_resps` are the outbound prep steps computed in the previous step.
 The order MUST match `AggregationJobInitReq.prepare_inits`.
 
-The Helper responds to the Leader with HTTP status code 201 Created and a body
-consisting of the `AggregationJobResp`, with media type
+After receiving the response to its `AggregationJobInitReq`, the Leader makes
+requests to
+`GET {helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}` to check
+on the status of the aggregation job. If the aggregation job is not finished
+yet, the Helper MAY wait for the job to complete. Once the aggregation job is
+ready, or if it is already ready, the Helper responds with HTTP status 200 OK
```

Comment:
Once a job is ready for collection, do _all_ polls have to return `200 OK` consistently?

I ask because the usage of eventually consistent mechanisms could lead to polls even after one which returned 200 OK sometimes indicating that the job isn't ready.

Reply:
I don't think a poll has to return 200 OK consistently for the protocol to work. Extra polls are harmless--the leader will simply try again. Or perhaps, whether it's harmless depends on the consistency guarantees between leader and helper, which I don't think DAP is prescriptive of right now.

I also don't think that behavior runs awry of the text. Perhaps we could disambiguate by saying `the Helper SHOULD respond with...` but that's a bit awkward since it implies a compliant aggregator could never return 200 OK... (To be clear, I think the behavior you describe should be allowable).

Reply:
I didn't think that behavior was proscribed by the text either, but wanted to verify.

It's probably worth calling this out as a possibility in the text as a heads up to leader node implementors, though I'm not sure why a leader would try and poll again after seeing a completion.

Reply:
+1 to calling this out. No normative language required, just something like "At least one GET will get the AggregationJobResp" in return.

Reply:
I've been staring at this for a while and am having difficulty succinctly writing it out, without it sounding confusing.

The closest I've gotten so far is to add a sentence `For each aggregation job init/continue, the Helper will respond at least once with HTTP status 200 OK.`, but I'm not sure that actually covers the scenario we've painted.

Also the more I stare at it the less I'm onboard with taking it. I don't think a Leader will needs to take any different action or encode any different logic to cover this case. The only situation I can think of where this case comes up is where the 200 response is lost in transit, the leader queries again and receives 202. In which case, the solution is to proceed normally until 200 is actually received.